src/pykeen/evaluation/classification_evaluator.py

```diff
@@ -27,19 +30,25 @@ class ClassificationMetricResults(MetricResults):
     metrics = CLASSIFICATION_METRICS
 
     @classmethod
-    def from_scores(cls, y_true, y_score):
+    def from_scores(cls, y_true: np.ndarray, y_score: np.ndarray):
         """Return an instance of these metrics from a given set of true and scores."""
         data = dict()
         for key, metric in CLASSIFICATION_METRICS.items():
+            if y_true.size == 0:
+                logger.warning("Empty y_true for %s?!", key)
+                continue
             value = metric.score(y_true, y_score)
             if isinstance(value, np.number):
                 # TODO: fix this upstream / make metric.score comply to signature
                 value = value.item()
             data[key] = value
-        return ClassificationMetricResults(data=data)
+        data["num_scores"] = y_score.size
```

Comment:
is this guaranteed an int or can it be a multiple dimensional tuple?

Reply:
It is guaranteed to be a scalar, cf. [here](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.size.html). The other property is named [`shape`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.shape.html).